aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthValidationTest.kt

```diff
@@ -0,0 +1,545 @@
+            idToken = dummyToken,
+            accessToken = dummyToken,
+            refreshToken = dummyToken,
+            expiration = 300
+        )
+        every { createSignOutUri() } returns mockk()
+    }
+
+    private val activity = mockk<Activity>()
+
+    private val environment = AuthEnvironment(
+        context = mockk(),
+        configuration = configuration,
+        cognitoAuthService = mockk {
+            every { cognitoIdentityClient } returns identityClient
+            every { cognitoIdentityProviderClient } returns identityProviderClient
+        },
+        credentialStoreClient = mockk(relaxed = true) {
+            coEvery { loadCredentials(any()) } returns AmplifyCredential.DeviceData(DeviceMetadata.Empty)
+        },
+        userContextDataProvider = null,
+        hostedUIClient = hostedUIClient,
+        logger = logger
+    )
+    private val stateMachine = AuthStateMachine(
+        environment,
+        initialState = AuthState.Configured(
+            authNState = AuthenticationState.SignedOut(signedOutData = SignedOutData()),
+            authZState = AuthorizationState.Configured()
+        )
+    )
+
+    private val plugin = RealAWSCognitoAuthPlugin(
+        configuration = configuration,
+        authEnvironment = environment,
+        authStateMachine = stateMachine,
+        logger = logger
+    )
+
+    private val mainThreadSurrogate = newSingleThreadContext("Main thread")
+
+//region Setup/Teardown
+
+    @Before
+    fun setup() {
+        Dispatchers.setMain(mainThreadSurrogate)
+
+        mockkStatic("com.amplifyframework.auth.cognito.AWSCognitoAuthSessionKt")
+        every { any<AmplifyCredential>().isValid() } returns true
+
+        mockkObject(AuthHelper)
+        coEvery { AuthHelper.getSecretHash(any(), any(), any()) } returns "a hash"
+    }
+
+    @After
+    fun teardown() {
+        Dispatchers.resetMain()
+        unmockkAll()
+    }
+
+//endregion
+
+//region SRP
+
+    // SRP 1
+    // Expected: Sign in and sign out successful
+    @Test
+    fun `SRP sign in existing user with correct password, SRP sign out`() {
+        signIn(username1, password1)
+        assertSignedInAs(username1)
+        signOut()
+        assertSignedOut()
+    }
+
+    // SRP 2
+    // Expected: Sign in with incorrect password fails
+    @Test
+    fun `SRP sign in existing user with correct password, SRP sign out, SRP sign in same user with incorrect password`() { // ktlint-disable max-line-length
+        signIn(username1, password1)
+        signOut()
+        assertSignedOut()
+        assertFails { signIn(username1, incorrectPassword) }
+    }
+
+    // SPR 3
+    // Expected: Each action is successful
+    @Test
+    fun `SRP sign in existing user with correct password, SRP sign out, SRP sign in different user with correct password`() { // ktlint-disable max-line-length
+        signIn(username1, password1)
+        signOut()
+        assertSignedOut()
+        signIn(username2, password2)
```

Comment:
I think it would be good to add an assert after this line that username2 is signed in.